### DISCUSSION TITLE
Why does process.env take precedence over .env files?

### DISCUSSION THREAD

**Discussion by ppbl** [prisma/prisma#21207] [Q&A]
## Problem

I set `DATABASE_URL` in the `.env` file of the project, I also set `DATABASE_URL` in the server's `.bashrc` (I know I probably shouldn't have set it), then run the project and Prisma reads the system environment variables instead of the environment variables set in the .env file within the project

## Suggested solution

Prisma should prioritize using the .env file within the project 

## Additional context

I used `prisma db push`, and due to this issue, I accidentally emptied the tables in the database pointed to by the system environment variable `DATABASE_URL`.😭

**Comment by ludralph** [comment]
Hi @ppbl 👋 

Thank you for raising this question.

Prisma always reads environment variables from the system's environment. When you use Prisma CLI or Prisma Client, the `.env` file content and the variables defined in it are added to the `process.env` object, where Prisma can read it and use it. However, if an environment variable is already set in the system's environment, it will take precedence over the same variable defined in the `.env` file. This is because of the way the `dotenv` package works, which Prisma uses under the hood to load environment variables from .env files.

According to the dotenv [documentation](https://www.npmjs.com/package/dotenv#what-happens-to-environment-variables-that-were-already-set):

> Variables are applied in the order they are present. If there is a collision,  then that variable in the `.env` file will be skipped. This means that system environment variables have higher priority and they will overwrite the same variables defined in `.env` files.
Therefore, if you have `DATABASE_URL` set in both your system's environment and in your `.env` file, the value from the system's environment will be used.
> If you want to use the value from the `.env` file, you would need to remove the `DATABASE_URL` variable from your system's environment.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
